src/index.ts

```diff
@@ -47,27 +47,36 @@ export const handleTokenRequest = async (ctx: Context, request: Request) => {
 	const key = await ctx.cache.ISSUANCE_KEYS.get(keyID);
 
 	if (key === null) {
-		throw new Error('Issuer not initialised');
+		ctx.metrics.issuanceKeyErrorTotal.inc({ key_id: keyID, type: KeyError.NOT_FOUND });
+		throw new Error('No key found for the requested key id');
 	}
 
 	const privateKey = key.data;
 	if (privateKey === undefined) {
-		throw new Error('Issuer not initialised');
+		ctx.metrics.issuanceKeyErrorTotal.inc({ key_id: keyID, type: KeyError.MISSING_PRIVATE_KEY });
+		throw new Error('No private key found for the requested key id');
+	}
+	let sk: CryptoKey;
+	try {
+		sk = await crypto.subtle.importKey(
+			'pkcs8',
+			privateKey,
+			{
+				name: ctx.isTest() ? 'RSA-PSS' : 'RSA-RAW',
+				hash: 'SHA-384',
+				length: 2048,
+			},
+			true,
+			['sign']
+		);
+	} catch (e) {
+		ctx.metrics.issuanceKeyErrorTotal.inc({ key_id: keyID, type: KeyError.INVALID_PRIVATE_KEY });
+		throw e;
 	}
-	const sk = await crypto.subtle.importKey(
-		'pkcs8',
-		privateKey,
-		{
-			name: ctx.isTest() ? 'RSA-PSS' : 'RSA-RAW',
-			hash: 'SHA-384',
-			length: 2048,
-		},
-		true,
-		['sign']
-	);
 	const pkEnc = key?.customMetadata?.publicKey;
 	if (!pkEnc) {
-		throw new Error('Issuer not initialised');
+		ctx.metrics.issuanceKeyErrorTotal.inc({ key_id: keyID, type: KeyError.MISSING_PUBLIC_KEY });
+		throw new Error('No public key found for the requested key id');
```

Comment:
This checking could happen early, before trying to import the private key.

Reply:
agree. I did not move it here. I placed the variable definition close to its usage, which is ok I think.